addons/account/models/account_report.py

```diff
@@ -280,6 +280,17 @@ def _validate_formula(self):
                     expression.report_line_id.display_name,
                 ))
 
+    @api.constrains('sequence')
+    def _validate_account_report_line(self):
+        # by default the sequence is 0 for the account report lines.
+        sequence_with_zero = self.report_id.line_ids.filtered(lambda l: l.sequence == 0)
+        if len(sequence_with_zero) == 1:
```

Comment:
So, when they're sorted by id, you don't care about the constraint ? Why not just sorting them all by (sequence, id) ?

Reply:
because of this https://github.com/odoo/odoo/pull/118341#issuecomment-1514383565